### DISCUSSION TITLE
Zephyr was unable to find the toolchain. Is the environment misconfigured?

### DISCUSSION THREAD

**Discussion by gsisk911** [zephyrproject-rtos/zephyr#39052] [Q&A]
I have attempted to follow the Adafruit instructions to get zephyr setup on macOS and on the zephyr website using GCC ARM Embedded toolchain (https://docs.zephyrproject.org/1.9.0/getting_started/getting_started.html#third-party-x-compilers) I believe there is an issue with the ZEPHYR_TOOLCHAIN_VARIANT, CROSS_COMPILE, and, TOOLCHAIN_HOME variables. 

here is what the 'west build' command returns 

-- west build: generating a build system
Including boilerplate (Zephyr base (cached)): /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /Users/gasisk/zephyrproject/zephyr/samples/basic/blinky
-- Zephyr version: 2.7.99 (/Users/gasisk/zephyrproject/zephyr), build: v2.7.0-rc3-187-g6cfc6fa8e1ba
-- Found west (found suitable version "0.11.1", minimum required is "0.7.1")
-- Board: adafruit_feather_stm32f405
-- Cache files will be written to: /Users/gasisk/Library/Caches/zephyr
-- Found dtc: /usr/local/bin/dtc (found suitable version "1.6.1", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (/usr/local/bin)
CMake Error at /Users/gasisk/zephyrproject/zephyr/cmake/compiler/gcc/generic.cmake:9 (message):
  Zephyr was unable to find the toolchain.  Is the environment misconfigured?

  User-configuration:

  ZEPHYR_TOOLCHAIN_VARIANT: gnuarmemb

  Internal variables:

  CROSS_COMPILE: /usr/local/bin/bin/arm-none-eabi-

  TOOLCHAIN_HOME: /usr/local/bin

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/generic_toolchain.cmake:42 (include)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:543 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  CMakeLists.txt:4 (find_package)

-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /Users/gasisk/.virtualenvs/zephyenv/bin/cmake -DWEST_PYTHON=/Users/gasisk/.virtualenvs/zephyenv/bin/python -B/Users/gasisk/zephyrproject/zephyr/build -S/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky -GNinja

___________________________________________________________________________________________________

It looks to me like the values of the aforementioned variables are wrong because those aren't the locations of the toolchain I downloaded via bash (Adafruit setup) or from the arm developer website (zephyr setup) I have modified my .bash_profile to no avail.

Sorry I'm new to this and only self taught so there are probably massive knowledge gaps I don't know about. 

TIA to anyone who can help me

Thanks, Griffin Sisk

**Comment by stephanosio** [comment]
The documentation you linked above is very outdated and obsolete. Please refer to the latest documentation:
https://docs.zephyrproject.org/latest/getting_started/toolchain_3rd_party_x_compilers.html

Have you set `GCCARMEMB_TOOLCHAIN_PATH`?

**Reply by gsisk911** [reply]
Yes, echo $GNUARMEMB_TOOLCHAIN_PATH returns
/Users/gasisk/gcc-arm-none-eabi-10.3-2021.07

(the correct location of the file)

I have made progress by doing west -pristine 

Now I am here

(zephyenv) gasisk@MacBook-Pro zephyr % west build -p  auto -b adafruit_feather_stm32f405 samples/basic/blinky_pwm
-- west build: generating a build system
Including boilerplate (Zephyr base): /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm
-- Zephyr version: 2.7.99 (/Users/gasisk/zephyrproject/zephyr), build: v2.7.0-rc3-187-g6cfc6fa8e1ba
-- Found Python3: /Users/gasisk/.virtualenvs/zephyenv/bin/python3.9 (found suitable exact version "3.9.7") found components: Interpreter 
-- Found west (found suitable version "0.11.1", minimum required is "0.7.1")
-- Board: adafruit_feather_stm32f405
-- Cache files will be written to: /Users/gasisk/Library/Caches/zephyr
-- Found dtc: /usr/local/bin/dtc (found suitable version "1.6.1", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (/Users/gasisk/gcc-arm-none-eabi-10.3-2021.07)
-- Found BOARD.dts: /Users/gasisk/zephyrproject/zephyr/boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405.dts
adafruit_feather_stm32f405.dts.pre.tmp:94.39-99.5: Warning (interrupt_provider): /soc/interrupt-controller@40013c00: Missing #address-cells in interrupt provider
-- Generated zephyr.dts: /Users/gasisk/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /Users/gasisk/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /Users/gasisk/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /Users/gasisk/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /Users/gasisk/zephyrproject/zephyr/Kconfig
Loaded configuration '/Users/gasisk/zephyrproject/zephyr/boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405_defconfig'
Merged configuration '/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm/prj.conf'
Configuration saved to '/Users/gasisk/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/Users/gasisk/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.1
-- The CXX compiler identification is GNU 10.3.1
-- The ASM compiler identification is GNU
-- Found assembler: /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-gcc
Including boilerplate (Zephyr base (cached)): /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm
CMake Error at /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:78 (add_custom_target):
  add_custom_target cannot create target "code_data_relocation_target"
  because another target with the same name already exists.  The existing
  target is a custom target created in source directory
  "/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


CMake Error at /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:99 (add_custom_target):
  add_custom_target cannot create target "runners_yaml_props_target" because
  another target with the same name already exists.  The existing target is a
  custom target created in source directory
  "/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


-- Zephyr version: 2.7.99 (/Users/gasisk/zephyrproject/zephyr), build: v2.7.0-rc3-187-g6cfc6fa8e1ba
-- Found west (found suitable version "0.11.1", minimum required is "0.7.1")
CMake Error at /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:185 (add_custom_target):
  add_custom_target cannot create target "pristine" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


CMake Warning at /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2378 (message):
  

     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable BOARD set to "adafruit_feather_stm32f405" in:
     *     /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: BOARD must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:9223372036854775807 (zephyr_variable_set_too_late)
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2332 (set)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:212 (zephyr_check_cache)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


CMake Warning at /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2378 (message):
  

     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable BOARD set to "adafruit_feather_stm32f405" in:
     *     /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: BOARD must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:9223372036854775807 (zephyr_variable_set_too_late)
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2353 (set)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:212 (zephyr_check_cache)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


CMake Warning at /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2378 (message):
  

     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable SHIELD set to "" in:
     *     /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: SHIELD must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:9223372036854775807 (zephyr_variable_set_too_late)
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2336 (unset)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:291 (zephyr_check_cache)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


CMake Warning at /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2378 (message):
  

     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable SHIELD set to "" in:
     *     /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: SHIELD must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:9223372036854775807 (zephyr_variable_set_too_late)
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2353 (set)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:291 (zephyr_check_cache)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


-- Board: adafruit_feather_stm32f405
CMake Warning at /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2378 (message):
  

     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable CONF_FILE set to "/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm/prj.conf" in:
     *     /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: CONF_FILE must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:9223372036854775807 (zephyr_variable_set_too_late)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:430 (set)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)


CMake Warning at /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2378 (message):
  

     **********************************************************************
     *
     *                    WARNING
     *
     * CMake variable CONF_FILE set to "/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky_pwm/prj.conf" in:
     *     /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
     *
     * This is too late to make changes! The change was ignored.
     *
     * Hint: CONF_FILE must be set before calling find_package(Zephyr ...).
     *
     **********************************************************************

Call Stack (most recent call first):
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:9223372036854775807 (zephyr_variable_set_too_late)
  /Users/gasisk/zephyrproject/zephyr/cmake/extensions.cmake:2200 (set)
  /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake:474 (zephyr_file)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /Users/gasisk/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:40 (include_boilerplate)
  /Users/gasisk/zephyrproject/zephyr/CMakeLists.txt:4 (find_package)

**Reply by stephanosio** [reply]
The `samples/basic/blinky_pwm` sample is not supported on the `adafruit_feather_stm32f405` board because it does not define `pwm-led0` alias in the devicetree.

https://github.com/zephyrproject-rtos/zephyr/blob/a8f02ed5eec27de81a2759fc7b95530c452bb862/samples/basic/blinky_pwm/sample.yaml#L6

Try building another sample (e.g. `samples/hello_world`).

As for the strange CMake errors, try the following (do each step and check if build works):

1. Nuke the build directory (run `rm -rf build` from where you invoked `west build`)
2. Nuke the cache directory (run `rm -rf /Users/gasisk/Library/Caches/zephyr`)
3. Re-export the Zephyr CMake package (run `west zephyr-export` from the root of the Zephyr source tree)
4. Set `ZEPHYR_BASE` manually (`export ZEPHYR_BASE=/Users/gasisk/zephyrproject/zephyr`)

**Reply by gsisk911** [reply]
Did all of what you said with no luck, then also performed git pull to try and refresh what I had locally and I now see these new errors...




(zephyenv) gasisk@MacBook-Pro zephyr % west build --pristine -b adafruit_feather_stm32f405 samples/basic/blinky
-- west build: generating a build system
Including boilerplate (Zephyr base): /Users/gasisk/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Application: /Users/gasisk/zephyrproject/zephyr/samples/basic/blinky
-- Zephyr version: 2.7.99 (/Users/gasisk/zephyrproject/zephyr), build: v2.7.0-rc3-240-ga8f02ed5eec2
-- Found Python3: /Users/gasisk/.virtualenvs/zephyenv/bin/python3.9 (found suitable exact version "3.9.7") found components: Interpreter 
-- Found west (found suitable version "0.11.1", minimum required is "0.7.1")
-- Board: adafruit_feather_stm32f405
-- Cache files will be written to: /Users/gasisk/Library/Caches/zephyr
-- Found dtc: /usr/local/bin/dtc (found suitable version "1.6.1", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (/Users/gasisk/gcc-arm-none-eabi-10.3-2021.07)
-- Found BOARD.dts: /Users/gasisk/zephyrproject/zephyr/boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405.dts
adafruit_feather_stm32f405.dts.pre.tmp:94.39-99.5: Warning (interrupt_provider): /soc/interrupt-controller@40013c00: Missing #address-cells in interrupt provider
-- Generated zephyr.dts: /Users/gasisk/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /Users/gasisk/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /Users/gasisk/zephyrproject/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /Users/gasisk/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /Users/gasisk/zephyrproject/zephyr/Kconfig
Loaded configuration '/Users/gasisk/zephyrproject/zephyr/boards/arm/adafruit_feather_stm32f405/adafruit_feather_stm32f405_defconfig'
Merged configuration '/Users/gasisk/zephyrproject/zephyr/samples/basic/blinky/prj.conf'
Configuration saved to '/Users/gasisk/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/Users/gasisk/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.3.1
-- The CXX compiler identification is GNU 10.3.1
-- The ASM compiler identification is GNU
-- Found assembler: /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/gasisk/zephyrproject/zephyr/build
-- west build: building application
[1/145] Preparing syscall dependency handling

[68/145] Linking C static library app/libapp.a
FAILED: app/libapp.a 
: && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E rm -f app/libapp.a && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc app/libapp.a  CMakeFiles/app.dir/src/main.c.obj && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ranlib app/libapp.a && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E touch app/libapp.a && :
/bin/sh: line 1: 88512 Killed: 9               /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc app/libapp.a CMakeFiles/app.dir/src/main.c.obj
[69/145] Linking C static library zephyr/libzephyr.a
FAILED: zephyr/libzephyr.a 
: && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E rm -f zephyr/libzephyr.a && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc zephyr/libzephyr.a  zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32f4/soc.c.obj zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/stm32cube_hal.c.obj zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/soc_config.c.obj zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj zephyr/CMakeFiles/zephyr.dir/drivers/pinmux/pinmux_stm32.c.obj && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ranlib zephyr/libzephyr.a && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E touch zephyr/libzephyr.a && :
/bin/sh: line 1: 88587 Killed: 9               /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc zephyr/libzephyr.a zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32f4/soc.c.obj zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/stm32cube_hal.c.obj zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/soc_config.c.obj zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj zephyr/CMakeFiles/zephyr.dir/drivers/pinmux/pinmux_stm32.c.obj
[70/145] Linking C static library zephyr/arch/common/libisr_tables.a
FAILED: zephyr/arch/common/libisr_tables.a 
: && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E rm -f zephyr/arch/common/libisr_tables.a && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc zephyr/arch/common/libisr_tables.a  zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ranlib zephyr/arch/common/libisr_tables.a && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E touch zephyr/arch/common/libisr_tables.a && :
/bin/sh: line 1: 88545 Killed: 9               /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc zephyr/arch/common/libisr_tables.a zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[71/145] Linking C static library zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a
FAILED: zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a 
: && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E rm -f zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj && /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ranlib zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a && /Users/gasisk/.virtualenvs/zephyenv/lib/python3.9/site-packages/cmake/data/CMake.app/Contents/bin/cmake -E touch zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a && :
/bin/sh: line 1: 88617 Killed: 9               /Users/gasisk/gcc-arm-none-eabi-10.3-2021.07/bin/arm-none-eabi-ar qc zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj
[85/145] Building C object zephyr/drivers/interrupt_con...drivers__interrupt_controller.dir/intc_exti_stm32.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /Users/gasisk/.virtualenvs/zephyenv/bin/cmake --build /Users/gasisk/zephyrproject/zephyr/build

**Reply by stephanosio** [reply]
Try `west update`.

**Reply by gsisk911** [reply]
Thank you so much! Now on to flashing the board!
